src/main.cpp

```diff
@@ -33,63 +37,136 @@ constexpr auto VERSION = _VERSION;
-               "[--client]"
-            << std::endl
-            << std::endl;
-
-  std::cout << "Multi-touch gesture recognizer." << std::endl;
-  std::cout << "Touchégg is an app that runs in the background and transform "
-               "the gestures you make on your touchpad into visible actions in "
-               "your desktop."
-            << std::endl;
-  std::cout << "For more information please visit:" << std::endl;
-  std::cout << "https://github.com/JoseExposito/touchegg" << std::endl
-            << std::endl;
-
-  std::cout << "Option\t\tMeaning" << std::endl;
-  std::cout << " --daemon\tRun Touchégg in daemon mode. This mode starts a "
-               "service that gathers gestures but executes no actions"
-            << std::endl;
-  std::cout << " --client\tConnect to an existing Touchégg daemon and "
-               "execute actions in your desktop"
-            << std::endl;
-  std::cout << "Without arguments Touchégg starts in client mode" << std::endl
-            << std::endl;
+  const std::string getCmdOption(const std::string& option) const {
+    std::vector<std::string>::const_iterator itr;
+    itr = std::find(this->tokens.begin(), this->tokens.end(), option);
+    if (itr != this->tokens.end() && ++itr != this->tokens.end()) {
+      return *itr;
+    }
+    static const std::string empty_string("");
+    return empty_string;
+  }
+
+  // special case used for daemon thresholds
+  void getCmdOption2d(const std::string& option, double& d1, double& d2) const {
+    std::vector<std::string>::const_iterator itr;
+    itr = std::find(this->tokens.begin(), this->tokens.end(), option);
+    if (itr != this->tokens.end()) {
+      if (++itr != this->tokens.end()) {
+        d1 = std::stod(*itr);
+        if (++itr != this->tokens.end()) {
+          d2 = std::stod(*itr);
+        }
+      }
+    }
+  }
+
+  bool cmdOptionExists(const std::string& option) const {
+    return std::find(this->tokens.begin(), this->tokens.end(), option) !=
+           this->tokens.end();
+  }
+
+ private:
+  std::vector<std::string> tokens;
+};
+
+// Parse the command line arguments
+void parseArgs(int argc, char** argv, bool& daemonMode, bool& clientMode,
+               double& startThreshold, double& finishThreshold) {
+  bool verbose, quiet, noGestures, noUpdates;
+
+  InputParser input(argc, argv);
+
+  // daemon takes precedence over client
+  if (input.cmdOptionExists("--daemon")) {
+    daemonMode = true;
+    input.getCmdOption2d("--daemon", startThreshold, finishThreshold);
+  } else if (input.cmdOptionExists("--client") || (argc == 1)) {
+    clientMode = true;
+  }
+
+  verbose = input.cmdOptionExists("--verbose") || input.cmdOptionExists("-v");
+  quiet = input.cmdOptionExists("--quiet") || input.cmdOptionExists("-q");
+  noGestures = input.cmdOptionExists("--no-gesture-messages");
+  noUpdates = input.cmdOptionExists("--no-update-messages");
+
+  // init Logger options
+  Logger::obj(verbose, quiet, noGestures, noUpdates);
```

Comment:
Initializing the logger here couples the argument parsing with the logger. It should be better to move it to the main method (see bellow)

Reply:
Of course separation of concerns & loose coupling is important. I did it this way because the 4 params to init Logger are only used once,  and I dislike passing too many params in/out of a func. Not a big deal in this case (4 more + moving the vars to main).

Reply:
After looking over the code again & thinking about it I agree with your suggestion.